src/app/youtube/components/catalog/catalog.component.ts

```diff
@@ -10,29 +17,45 @@ import { SortbarManagerService } from '../../services/sortbar-manager.service'
   styleUrls: ['./catalog.component.scss']
 })
 export class CatalogComponent implements OnInit, OnDestroy {
-  videos: IVideoStatsItem[] = []
+  youtubeVideos: IVideoStatsItem[] = []
+
+  customVideos: IVideoCard[] = []
+
+  youtubeVideos$ = this.store.select(apiVideos)
+
+  customVideos$ = this.store.select(customVideos)
 
   videosSubscription!: Subscription
 
+  customVideosSubscription!: Subscription
+
   sortOptionsSubscription!: Subscription
 
   constructor(
     private searchService: SearchService,
-    private sortService: SortbarManagerService
+    private sortService: SortbarManagerService,
+    private store: Store<AppState>
   ) {}
 
   ngOnInit(): void {
-    this.videos = this.searchService.videoItems
-    this.videosSubscription = this.searchService.videos$.subscribe(data => {
-      this.videos = data
+    this.videosSubscription = this.youtubeVideos$.subscribe(videos => {
+      this.youtubeVideos = videos
+    })
+    this.customVideosSubscription = this.customVideos$.subscribe(videos => {
+      this.customVideos = videos
     })
     this.sortOptionsSubscription = this.sortService.options$.subscribe(() => {
-      this.videos = [...this.videos]
+      this.youtubeVideos = [...this.youtubeVideos]
     })
   }
 
   ngOnDestroy(): void {
     this.videosSubscription.unsubscribe()
+    this.customVideosSubscription.unsubscribe()
     this.sortOptionsSubscription.unsubscribe()
```

Comment:
here you can look for 'rxjs unsubscribe patterns angular', cause imagine how it'll look like when there's not 3, but a lot of observables to unsubscribe.